runtime/mangata-kusama/src/lib.rs

```diff
@@ -1989,6 +1991,84 @@ impl_runtime_apis! {
 					).unwrap_or_default()
 			}
 		}
+
+		fn is_sell_asset_lock_free(
+			path: Vec<TokenId>,
+			input_amount: Balance,
+			) -> Option<bool>{
+			match (path.len(), pallet_fee_lock::FeeLockMetadata::<Runtime>::get()) {
+				(length, _) if length < 2 => {
+					None
+				}
+				(2, Some(feelock)) => {
+					let input = path.get(0)?;
+					let output = path.get(1)?;
+					let output_amount = Xyk::calculate_sell_price_id(*input, *output, input_amount).ok()?;
+					Some(
+					FeeHelpers::<
+								Runtime,
+								orml_tokens::MultiTokenCurrencyAdapter<Runtime>,
+								ToAuthor,
+								OnChargeTransactionHandler,
+								FeeLock,
+								>::is_high_value_swap(&feelock, *input, input_amount)
+									||
+					FeeHelpers::<
+								Runtime,
+								orml_tokens::MultiTokenCurrencyAdapter<Runtime>,
+								ToAuthor,
+								OnChargeTransactionHandler,
+								FeeLock,
+								>::is_high_value_swap(&feelock, *output, output_amount)
+								)
+				}
+				(_,  None) => {
+					Some(false)
+				}
+				(_,  Some(_)) => {
+					Some(true)
+				}
+			}
+		}
+
+		fn is_buy_asset_lock_free(
+			path: Vec<TokenId>,
+			input_amount: Balance,
+			) -> Option<bool>{
+			match (path.len(), pallet_fee_lock::FeeLockMetadata::<Runtime>::get()) {
```

Comment:
I think this conditioning should be inside the pallet so we don't have it duplicated. Can we make such a function in the FeeLock pallet?

Reply:
actually at the moment xyk & feelock pallets are not aware of each other, its only in runtime where informations from both pallet are utilized. ideally it should be plased somewhere in common runtime module so both kusama & rococo could use it, maybe we should consider another look on that :thinking:

Reply:
Aaah, I see. Yes totally agree. Created ticket for it (MGX-534)